lib/spack/docs/getting_started.rst

```diff
@@ -473,6 +477,251 @@ Fortran.
+         f77:
+         fc:
+       spec: clang@5.0.0
+       target: x86_64
+   - compiler:
+       environment: {}
+       extra_rpaths: []
+       flags: {}
+       modules: []
+       operating_system: fedora25
+       paths:
+         cc: /opt/spack/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/llvm-3.9.0-66qwczd6rumppz6qmzlxstuyhkajxa3i/bin/clang
+         cxx: /opt/spack/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/llvm-3.9.0-66qwczd6rumppz6qmzlxstuyhkajxa3i/bin/clang++
+         f77:
+         fc:
+       spec: clang@3.9.0
+       target: x86_64
+
+In order to ensure that everything builds properly when specifying say
+``clang@5.0.0`` as the compiler, we will want to make sure that we modify the
+``modules: []`` list to contain the module that Spack has already generated.
+
+To find the correct module, which has already been generated by way of
+installing ``llvm``, we can perform the following:
+
+.. code-block:: console
+
+   # Recall that $SPACK_ROOT must be set, and in this example
+   # it is set to `/opt/spack`
+   #
+   # Recall also that your OS is different vvvvvvvvvvvvvvvvvvvvvv
+   $ ls -1 $SPACK_ROOT/share/spack/modules/linux-fedora25-x86_64/llvm-*
+   /opt/spack/share/spack/modules/linux-fedora25-x86_64/llvm-3.9.0-gcc-6.4.1-66qwczd
+   /opt/spack/share/spack/modules/linux-fedora25-x86_64/llvm-5.0.0-gcc-6.4.1-id4ljxr
+   # NOTE: gcc@6.4.1 is the compiler used to compile llvm          ^^^^^^^^^
+   #       That section, as well as the short hashes will be different for you.
+
+So for each new LLVM installation, we want to ensure that Spack will load the
+correct module before trying to compile any packages with it.
+
+.. tip::
+
+   ``modules: []`` is an empty YAML list.  We only need to add one module at
+   this point, but we will make a list.  Make sure that the hyphen (``-``)
+   is **directly** underneath the ``m`` of ``modules:``.
+
+**For** ``llvm@5.0.0``
+
+    .. code-block:: yaml
+
+       modules:
+       - llvm-5.0.0-gcc-6.4.1-id4ljxr
+
+**For** ``llvm@3.9.0``
+
+    .. code-block:: yaml
+
+       modules:
+       - llvm-3.9.0-gcc-6.4.1-66qwczd
+
+Now Spack will load the compiler's module before trying to compile any packages.
+
+"""""""""""""""""""""""""""""""""
+Step 4: Choose a Fortran Compiler
+"""""""""""""""""""""""""""""""""
+
+Since LLVM does not currently provide a Fortran compiler, you will want to make
+sure that you have one configured (many of the core HPC packages need one).  In
+this example, since Fedora 25 comes with a modern Fortran compiler via its
+package manager, this was used:
+
+.. code-block:: yaml
+
+   paths:
+     cc: /opt/spack/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/llvm-5.0.0-id4ljxr45ihmyys2qr76ce7o7ungrtag/bin/clang
+     cxx: /opt/spack/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/llvm-5.0.0-id4ljxr45ihmyys2qr76ce7o7ungrtag/bin/clang++
+     #    vvvvvvvvvvvvvvvvv
+     f77: /usr/bin/gfortran
```

Comment:
Unfortunately this currently may not always work on Linux, reason being that compiler wrappers have hard-coded names and some packages like openblas rely on those names being right.

In the current develop of Spack fortran names in Clang is flang (unless the PR which changed that isn’t merged yet)

Reply:
Maybe @tgamblin could add dynamic symlinks in compiler wrappers

Reply:
I don't think I understand.  I used `/usr/bin/gfortran` for everything I just did for this desktop, started about 4 days ago.  Are you saying placing `flang` in there would not work, or `gfortran` will not?

Reply:
Look at linked_paths variable here: https://github.com/LLNL/spack/blob/develop/lib/spack/spack/compilers/clang.py

Spack never uses compilers directly, only through wrappers. So any application compiled with Spack won’t see /use/bin/gfortran, instead FC/CC and alike will point to one of the wrappers.

As I said, this might be a problem only for some packages

Reply:
Here’s flang wrapper which points to the main wrapper which actually does the job based on environment/settings: https://github.com/LLNL/spack/blob/develop/lib/spack/env/clang/flang

Reply:
The application will still use /usr/bin/gfortran to build, but with the current Spack code it will do so through a file named “flang” regardless of what is put as f77 in clang compilers

Reply:
I think I'm really missing the point here.  The compiler wrappers wrap around whatever is specified as `fc` and `f77` in `compilers.yaml` (p.s., why no `f90`?).  So regardless of what the compiler wrapper is called, as long as it wraps what is in `compilers.yaml`, it all checks out right?

Reply:
> So regardless of what the compiler wrapper is called, as long as it wraps what is in compilers.yaml, it all checks out right?

not necessarily. We had issues long time ago with `openblas` which would not allow to build with a fortran compiler wrapper named `f77`, it really had to be meaningful. See https://github.com/LLNL/spack/pull/949/files

Reply:
Good god.  So I guess the question then becomes does the tutorial remain as is, or should an admonition be added.

Argument for leaving as is: `openblas` compiled just fine for me using `/usr/bin/gfortran`.

Argument against: if it doesn't work for people, it's kind of a bait-and-switch.  "Here's *exactly* how you do it."  Then it doesn't work.  And we've misled them.  What is the fix though?

Reply:
> as is: openblas compiled just fine for me using /usr/bin/gfortran.

if that's the case, keep it as-is. 

> What is the fix though?

last paragraph here https://github.com/LLNL/spack/pull/5503#discussion_r142002265